Nos/Service/Relay/PagedRelaySubscription.swift

```diff
@@ -73,13 +74,18 @@ class PagedRelaySubscription {
                           
                     newUntilDates[subscription.relayAddress] = newDate
                     await subscriptionManager.decrementSubscriptionCount(for: subscriptionID)
+                    subscriptionsToRemove.append(subscription)
                 }
             }
             
+            subscriptionsToRemove.forEach { subscription in
```

Comment:
I really like when the solution involves using the right data structure. Given that we are already switching to a Set, how about making subscriptionsToRemove a Set of IDs, so that this line I'm commenting can be just `pagedSubscriptionIDs.subtract(subscriptionsToRemove)`?

Reply:
oh yeah, great idea!